src/main/java/com/teamabnormals/abnormals_core/common/network/MessageS2CServerRedirect.java

```diff
@@ -0,0 +1,55 @@
+package com.teamabnormals.abnormals_core.common.network;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.ConnectingScreen;
+import net.minecraft.client.gui.screen.DirtMessageScreen;
+import net.minecraft.client.gui.screen.MainMenuScreen;
+import net.minecraft.client.gui.screen.MultiplayerScreen;
+import net.minecraft.client.multiplayer.ServerData;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.network.NetworkEvent;
+
+import java.util.function.Supplier;
+
+/**
+ * Allows servers to redirect clients to another server.
+ * @author Jackson
+ */
+public class MessageS2CServerRedirect {
```

Comment:
Perhaps this class should be `final` since it isn't something intended to be able to be extended.